src/Runner.Common/JobServerQueue.cs

```diff
@@ -93,6 +98,20 @@ public void Start(Pipelines.AgentJobRequestMessage jobRequest)
 
             _jobServer.InitializeWebsocketClient(serviceEndPoint);
 
+            // This code is usually wrapped by an instance of IExecutionContext which isn't available here.
+            jobRequest.Variables.TryGetValue("system.github.results_endpoint", out VariableValue resultsEndpointVariable);
```

Comment:
should we use the SystemConnection to pass the URL? we have the same pattern for passing `CacheServerUrl` and `GenerateIdTokenUrl` from the service to the runner.

Reply:
What's the difference and the advantage of using `SystemConnection`?

Reply:
Going to keep is as a variable for now.  The `SystemConnction` has authorization parameters and for Results we are just sticking with the system access token.

Reply:
the auth in the `SystemConnection` is exactly the system access token.
the advantage is aline with the existing pattern, the `SystemConnection` was holding some connection info back to the service, I think the result service URL is just one of them.